internal/lsp/documents_store.go

```diff
@@ -0,0 +1,28 @@
+package lsp
+
+import "sync"
+
+type documentStore[Doc any] struct {
```

Comment:
I'm aware that a `sync.Map` data structure exists, but this one looks more "fine grained", and its simple enough